agent/acs/client/acs_client_test.go

```diff
@@ -49,146 +52,75 @@ const sampleCredentialsMessage = `
 }
 `
 
-type messageLogger struct {
-	writes [][]byte
-	reads  [][]byte
-	closed bool
-}
+const (
+	TestClusterArn  = "arn:aws:ec2:123:container/cluster:123456"
+	TestInstanceArn = "arn:aws:ec2:123:container/containerInstance/12345678"
+)
 
 var testCfg = &config.Config{
 	AcceptInsecureCert: true,
 	AWSRegion:          "us-east-1",
 }
 
-func (ml *messageLogger) WriteMessage(_ int, data []byte) error {
-	if ml.closed {
-		return errors.New("can't write to closed ws")
-	}
-	ml.writes = append(ml.writes, data)
-	return nil
-}
-
-func (ml *messageLogger) Close() error {
-	ml.closed = true
-	return nil
-}
-
-func (ml *messageLogger) ReadMessage() (int, []byte, error) {
-	for len(ml.reads) == 0 && !ml.closed {
-		time.Sleep(1 * time.Millisecond)
-	}
-	if ml.closed {
-		return 0, []byte{}, errors.New("can't read from a closed websocket")
-	}
-	read := ml.reads[len(ml.reads)-1]
-	ml.reads = ml.reads[0 : len(ml.reads)-1]
-	return websocket.TextMessage, read, nil
-}
-
-func testCS() (wsclient.ClientServer, *messageLogger) {
-	testCreds := credentials.AnonymousCredentials
+func TestMakeUnrecognizedRequest(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
 
-	cs := New("localhost:443", testCfg, testCreds).(*clientServer)
-	ml := &messageLogger{make([][]byte, 0), make([][]byte, 0), false}
-	cs.SetConnection(ml)
-	return cs, ml
-}
+	conn := mock_wsclient.NewMockWebsocketConn(ctrl)
+	conn.EXPECT().Close()
 
-func TestMakeUnrecognizedRequest(t *testing.T) {
-	cs, _ := testCS()
+	cs := testCS(conn)
+	defer cs.Close()
 	// 'testing.T' should not be a known type ;)
 	err := cs.MakeRequest(t)
 	if _, ok := err.(*wsclient.UnrecognizedWSRequestType); !ok {
 		t.Fatal("Expected unrecognized request type")
 	}
-	_ = err.Error() // This is one of those times when 100% coverage is silly
-	cs.Close()
-}
-
-func strptr(s string) *string {
-	return &s
 }
 
 func TestWriteAckRequest(t *testing.T) {
-	cs, ml := testCS()
-
-	req := ecsacs.AckRequest{Cluster: strptr("default"), ContainerInstance: strptr("testCI"), MessageId: strptr("messageID")}
-	err := cs.MakeRequest(&req)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	write := ml.writes[0]
-	writtenReq := struct {
-		Type    string
-		Message ecsacs.AckRequest
-	}{}
-	err = json.Unmarshal(write, &writtenReq)
-	if err != nil {
-		t.Fatal("Unable to unmarshal written", err)
-	}
-	msg := writtenReq.Message
-	if *msg.Cluster != "default" || *msg.ContainerInstance != "testCI" || *msg.MessageId != "messageID" {
-		t.Error("Did not write what we expected")
-	}
-	cs.Close()
-}
-
-func TestPayloadHandlerCalled(t *testing.T) {
```

Comment:
I removed this test because it looked like it was a similar but less thorough version of the credentials handler test (same code path, different struct moving through it).
However, I don't want to be the guy that shows up and starts deleting all of the tests  &mdash; if I'm missing or overlooking a case here please let me know.